community_server/config/routes.php

```diff
@@ -65,7 +65,7 @@
             if($entry == 'ElopageWebhook' || $entry == 'AppRequests') {
               return true;
             }
-            if($request->clientIp() == '127.0.0.1' || $request->clientIp() == 'localhost') {
+            if($request->clientIp() == '127.0.0.1' || $request->clientIp() == 'localhost' || $request->clientIp() == '') {
               return true;
             }
```

Comment:
I do not understand this block. Handling localhost differently might cause problems since the test & development environment is on localhost - other setups might not.

Reply:
The function in which this code is located ($csrf->whitelistCallback)
check if csrf check can be disabled for this call. 
I use this to only allow ajax requests from localhost and from ips in config/app.php in API.allowedCaller array. 
When I running tests via Unit Tests (build-in in cakephp) clientIp return empty string instead of expected localhost or 127.0.0.1 so I have added this.

Reply:
👍🏻